jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/LayerSpec.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.cli.buildfile;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import java.io.IOException;
+
+/**
+ * Polymorphic yaml LayerSpec interface with custom deserializer, can parse both {@link
+ * ArchiveLayerSpec} and {@link FileLayerSpec}.
+ */
+@JsonDeserialize(using = LayerSpec.Deserializer.class)
+public interface LayerSpec {
+  class Deserializer extends StdDeserializer<LayerSpec> {
```

Comment:
Probably `LayerSpec.Deserializer` is publicly accessible, but I guess it doesn't matter because all the buildfile classes are internal and not public API, right?

Reply:
That's a good point, but since interfaces don't allow the private modifier, I think this is fine. Otherwise we would have to add another class to do this?